stan/math/rev/fun/unit_vector_constrain.hpp

```diff
@@ -88,11 +58,11 @@ Eigen::Matrix<var, R, C> unit_vector_constrain(
  * @return Unit length vector of dimension K
  * @param lp Log probability reference to increment.
  **/
-template <int R, int C>
-Eigen::Matrix<var, R, C> unit_vector_constrain(
-    const Eigen::Matrix<var, R, C>& y, var& lp) {
-  Eigen::Matrix<var, R, C> x = unit_vector_constrain(y);
-  lp -= 0.5 * dot_self(y);
+template <typename T, require_eigen_vt<is_var, T>* = nullptr>
+inline auto unit_vector_constrain(const T& y, var& lp) {
+  const auto& y_ref = to_ref(y);
+  auto x = unit_vector_constrain(y_ref);
+  lp -= 0.5 * dot_self(y_ref);
```

Comment:
[optional] We should look at the constrains in the future so we aren't doing things like this dot_self separately. I think this is the version of the function that gets used in Stan models.

Reply:
I say this as optional because we will be back through this code in the very short future for `var<mat>` and so I'm not really worried about getting it now. Fixing the includes and turning on this file is enough of a performance bump.

Reply:
Did not do.